ios/engine/KMEI/KeymanEngine/Classes/Model/LanguageResource.swift

```diff
@@ -25,6 +25,8 @@ public protocol LanguageResourceFullID {
 public protocol AnyLanguageResource {
   var id: String { get }
   var languageID: String { get }
+  // Was not always tracked within KeymanEngine - is optional for legacy reasons.
+  var packageID: String? { get }
```

Comment:
On Android, the legacy keyboards that don't have a package ID fall back to `cloud` for the package ID.
This way, you can always namespace `stub.KP + "::" + stub.KI`

The legacy keyboards also reside in a folder `cloud/` so we know not to parse a kmp.json file there. (And we know the keyboard version is in the filename).

Reply:
Yeah, iOS won't be doing that.  Instead, iOS will make a "KMP" (really, a fill-in `kmp.json` and dedicated folder) for the cloud resource.

Reply:
See #3212 for the "fill-in" code.  That PR doesn't write it out quite yet, but all the pieces are ready to do exactly that.

Reply:
ic.

Is `packageID` required for lexical models, then?

Reply:
Not at this time, since lexical model packages only contain just the one, right?  Is there lexical model namespacing in the Android engine?

Reply:
> since lexical model packages only contain just the one, right?

Correct

> Is there lexical model namespacing in the Android engine?

No for KMEA. `packageID` is only used for obtaining the file path.